api/all/src/main/java/io/opentelemetry/api/baggage/propagation/Parser.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.baggage.propagation;
+
+import io.opentelemetry.api.baggage.BaggageBuilder;
+import io.opentelemetry.api.baggage.BaggageEntryMetadata;
+
+/**
+ * Implements single-pass Baggage parsing in accordance with https://w3c.github.io/baggage/ Key /
+ * value are restricted in accordance with https://www.ietf.org/rfc/rfc2616.txt
+ *
+ * <p>Note: following aspects are not specified in RFC: - some invalid elements (key or value) -
+ * parser will include valid ones, disregard invalid - empty "value" is regarded as invalid - meta -
+ * anything besides element terminator (comma)
+ */
+class Parser {
+
+  private enum State {
+    KEY,
+    VALUE,
+    META
+  }
+
+  private final String baggageHeader;
+
+  private final Element key =
+      new Element(
+          new char[] {
+            '(', ')', '<', '>', '@', ',', ';', ':', '\\', '"', '/', '[', ']', '?', '=', '{', '}'
+          });
+  private final Element value = new Element(new char[] {'"', ',', ';', '\\'});
+  private String meta;
+
+  private State state;
+  private int metaStart;
+
+  private boolean skipToNext;
+
+  public Parser(String baggageHeader) {
+    this.baggageHeader = baggageHeader;
+    reset(0);
+  }
+
+  public void reset(int i) {
+
```

Comment:
there's some odd extra whitespace in here.  Also, please go through this class and make everything private that can be private.

Reply:
Will do.